problem-29.c

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <stdlib.h>
+// Uses structures to show names, roll numbers and percentages of 5 students but stores and reads content from text file
+
+typedef struct
+{
+    char *name;
+    int roll_number;
+    float percentage;
+} Student;
+
+int main()
+{
+    FILE *file;
+    Student data[5];
+
+    printf("Enter data of 5 students:\n\n");
+
+    for (int i = 0; i < 5; i++)
+    {
+        printf("\nStudent %d\n", i + 1);
+
+        printf("Name: ");
+        scanf("%s", &data[i].name);
+
+        printf("Roll Number: ");
+        scanf("%d", &data[i].roll_number);
+
+        printf("Percentage: ");
+        scanf("%f", &data[i].percentage);
+    }
+
+    file = fopen("text-files/student-data.txt", "w");
+
+    for (int i = 0; i < 5; i++)
+    {
+        fprintf(file, "%s %d %.1f\n", data[i].name, data[i].roll_number, data[i].percentage);
+    }
+
+    fclose(file);
+
+    file = fopen("text-files/student-data.txt", "r");
+
+    if (file == NULL)
+    {
+        printf("Failed to open file");
+        return 1;
+    }
+
+    Student students[5];
+
+    for (int i = 0; i < 5; i++)
+    {
+        students[i].name = malloc(50 * sizeof(char));

```

Comment:
What will happen if name is greater than 50 characters? Is it better to write each value in a separate line?